compiler/src/service/driver.rs

```diff
@@ -98,277 +96,43 @@ impl<W: World + WorkspaceProvider> Compiler for CompileDriverImpl<W> {
 
         self._relevant(event).unwrap_or(true)
     }
+
+    fn iter_dependencies<'a>(&'a self, f: &mut dyn FnMut(&'a ImmutPath, instant::SystemTime)) {
```

Comment:
how about some generic function that impl `FnMut(&'a ImmutPath, instant::SystemTime)`. You already made pretty heavy use of generics